### PR TITLE
docs(troubleshooting): Add troubleshooting section to docs

### PR DESCRIPTION
**Pull request by callumhemming**
# Description

Relating to issue: https://github.com/vendure-ecommerce/vendure/issues/2858 I'm adding a troubleshooting section to the README.md for common issues that pop up during installation and I am adding the issue I came across with some troubleshooting tips to deal with it. 

# Breaking changes

No

# Screenshots

image

# Checklist

📌 Always:
- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have [checked my own PR](## "Fix typo's and remove unused or commented out code")

👍 Most of the time:
- [ ] I have added or updated test cases
- [x] I have updated the README if needed

### CODE REVIEW

**Comment by michaelbromley** [comment]
Thanks! Out of interest which of the solutions worked for you?

**Comment by callumhemming** [comment]
> Thanks! Out of interest which of the solutions worked for you?

For me personally (running WSL and using pnpm), the third one(configuring the platform and installing sharp manually) was the only one that worked. In the issues thread users got it working by updating their package manger (yarn) and checking the version of node is supported seems to work best when this issue arises in the sharp repo. I tried a bunch of node version and no difference, I have a feeling using WSL is throwing sharp off when picking the correct binary :)

File: packages/create/templates/readme.hbs [hunk 1]
```diff
@@ -117,6 +117,15 @@ can set `dbConnectionOptions.synchronize` to `true`. This will cause the databas
...
+
+### Error: Could not load the "sharp" module using the \[OS\]-x\[Architecture\] runtime when running Vendure server.
+
+- Make sure your Node version is ^18.17.0 || ^20.3.0 || >=21.0.0 to support the Sharp library.
+- Make sure your package manager is up to date.
+- **Not recommended**: if none of the above helps to resolve the issue, install sharp specifying your machines OS and Architecture. For example: `pnpm install sharp --config.platform=linux --config.architecture=x64` or `npm install sharp --os linux --cpu x64`
+
 ---
 
 You can also run any pending migrations manually, without starting the server via the "vendure migrate" command.
```
Comment (michaelbromley) [inline packages/create/templates/readme.hbs]: This paragraph should be part of the "migrations" section

**Review by michaelbromley — CHANGES_REQUESTED** [review]

**Comment by michaelbromley** [comment]
OK. I just noticed that the troubleshooting section split the last paragraph of the "migrations" section. Can you rectify that and then it'd good to merge.

File: packages/create/templates/readme.hbs (hunk 1 above)
Comment (callumhemming) [reply packages/create/templates/readme.hbs]: Oops sorry! Fixed now

**Comment by callumhemming** [comment]
Just pushed a fix :D

**Comment by michaelbromley** [comment]
Thank you!

**Pull request merged**